vmdb/app/models/vim_performance_daily.rb

```diff
@@ -59,8 +59,24 @@ def self.find_by_time_profile(*args)
     #
 
     klass = ext_options[:class] || MetricRollup
-    options[:conditions] = klass.merge_conditions(options[:conditions], {:time_profile_id => tp_ids, :capture_interval_name => 'daily'})
-    klass.find(cnt, options)
+    wheres = klass.merge_conditions(options[:conditions], {:time_profile_id => tp_ids, :capture_interval_name => 'daily'})
+
+    # FIXME we should push these up to the callers
+    options.delete :conditions
+    select = options.delete :select
+    order = options.delete :order
+    raise "Unsupported options #{options.keys}" unless options.empty?
+
+    case cnt
+    when :all
+      scope = klass
+      scope = scope.select(select) if select
+      scope = scope.order(order) if order
+      scope = scope.where(wheres) if wheres
+      scope.all
+    else
+      raise "Unsupported finder value #{cnt}"
+    end
```

Comment:
Can we make a generic method that is shared by every activerecord subclass?  Or a module that does this kind of like @tenderlove did MiqPreloader?  I know you have the FIXME there, but it doesn't seem much of a stretch to extract it now.  I know @tenderlove had written that apply_options method elsewhere (can't remember where), and that has very similar logic.

Additionally, we should be careful with deleting from the options.  The callers may not be expecting that.  Not sure if options is passed in or not, so that may not apply here, but if we move to a module or mixin then we would have to be aware of that.  We can probably use a method on the hash or the keys like .except, .slice, .include_any? or something like that to simplify the validation without deleting it.